### PR TITLE
Replace LibLog with Microsoft.Extensions.Logging.Abstractions

### LINKED ISSUE #36: Replace LibLog with MS.Ext.Logging
LibLog is deprecated with Microsoft.Extensions.Logging being the recommend replacement.

### PR DESCRIPTION
**Pull request by damianh**
Static instances of LibLog ILog have been replaced with with instances that are dependency injected (either constructor or method param).

Fixes #36

### CODE REVIEW

**Comment by damianh** [comment]
@osoykan Would appreciate a review on this one as it contains some breaking changes. No rush at all on this one.

**Comment by damianh** [comment]
If you want a CI package to try it out, it'll need to be merged (can't produce CI packages from Github PRs because no access to secrets).

We can put the 1.x on to a maintenance release branch if we do merge this so can support bug fixes from that.

**Comment by osoykan** [comment]
It seems only breaking change adding a dependency to the `Listen` method since we call it after container built it wouldn't be a problem as resolving and passing the factory, I guess.

File: src/LittleForker/CooperativeShutdown.cs [hunk 1]
```diff
@@ -30,15 +28,19 @@ public static class CooperativeShutdown
...
         /// </param>
+        /// <param name="loggerFactory">
+        ///     A logger factory.
+        /// </param>
         /// <param name="onError">A method to be called if an error occurs while listening</param>
         /// <returns>
         ///     A disposable representing the named pipe listener.
         /// </returns>
-        public static Task<IDisposable> Listen(Action shutdownRequested, Action<Exception> onError = default)
+        public static Task<IDisposable> Listen(Action shutdownRequested, ILoggerFactory loggerFactory, Action<Exception> onError = default)
```
Comment (osoykan) [inline src/LittleForker/CooperativeShutdown.cs:38]: Logging might be optional maybe?
`loggerFactory ??= NullLoggerFactory.Instance;`

**Review by osoykan — APPROVED** [review]

File: src/LittleForker/ProcessSupervisor.cs [hunk 2]
```diff
@@ -60,27 +61,35 @@ private enum Trigger
...
         {
+            _loggerFactory = loggerFactory;
             _workingDirectory = workingDirectory;
             _processPath = processPath;
             _arguments = arguments ?? string.Empty;
             _environmentVariables = environmentVariables;
             _captureStdErr = captureStdErr;
 
-            _logger = LogProvider.GetLogger($"ProcessSupervisor-{processPath}");
+            _logger = loggerFactory.CreateLogger($"ProcessSupervisor-{processPath}");
```
Comment (cristiango) [inline src/LittleForker/ProcessSupervisor.cs]: What do you think about having all the logs outputted from LittleForker to have `SourceContext` starting with `LittleForker.xxxx`. This seems to be the standard also on Microsoft libs and allows us to control better where how the logs are distributed.
In Serilog, for example, we can use:
```csharp
new LoggerConfiguration().Filter.ByExcluding(Matching.FromSource("LittleForker"))
```
Or overriding minimum level
```csharp
new LoggerConfiguration().MinimumLevel.Override("LittleForker", LogEventLevel.Warning)
```


```suggestion
            _logger = loggerFactory.CreateLogger($"LittleForker.ProcessSupervisor-{processPath}");
```

**Comment by osoykan** [comment]
I did some changes:
- Resolved conflicts
- Added `LittleForker.` to the logger names
- Bumped dependencies

File: src/LittleForker/CooperativeShutdown.cs (hunk 1 above)
Comment (damianh) [reply src/LittleForker/CooperativeShutdown.cs:38]: I'm ok either way. Do you specifically want this?

**Comment by cristiango** [comment]
Thanks for getting this in. 
Did test it in our project with serilog
```csharp
Log.Logger = new LoggerConfiguration()
                .MinimumLevel.Verbose()
                .MinimumLevel.Override("LittleForker", LogEventLevel.Warning)
```
We now can get only `warn` from LittleForker yey!
Can you also push this version to nuget.org?

**Pull request merged**